### PR TITLE
ci(release): use github app token for tag push

### PR DESCRIPTION
**Pull request by project-defiant**
## ✨ Context

This PR is continuation of testing release of the gentropy Docker image with specific tag, when release is triggered - for more detail check https://github.com/opentargets/issues/issues/3384

## 🚦 Before submitting

- [ ] Do these changes cover one single feature (one change at a time)?
- [ ] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you make sure there is no commented out code in this PR?
- [ ] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [ ] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [ ] Did you make sure the changes pass local tests (`make test`)?
- [ ] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

**Comment by project-defiant** [comment]
@ireneisdoomed, please check the https://github.com/opentargets/issues/issues/3384#issuecomment-2239174254

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -34,14 +34,15 @@ jobs:
           fetch-depth: 0
           repository: opentargets/gentropy
           token: ${{ secrets.GITHUB_TOKEN }}
+          persist-credentials: false
```
Comment (ireneisdoomed) [inline .github/workflows/release.yaml:37]: Is this added for security reasons or because it is needed?

**Review by ireneisdoomed — APPROVED** [review]
So if I understand correctly you are swapping the Github access tokens between the one provided by the action, which generates a token from the Github App, and the secret token created by default.

The difference between them is that the App token can extend the repo's permissions  and can be used to trigger workflows, which is why you use it to create the tag with the semantic release.

**Comment by project-defiant** [comment]
That is correct!

File: .github/workflows/release.yaml (hunk 1 above)
Comment (project-defiant) [reply .github/workflows/release.yaml:37]: https://github.com/actions/checkout/issues/485 - I have added it not to persist this token, as it's not the default one.

**Pull request merged**